src/group_theory/quaternion_group.lean

```diff
@@ -0,0 +1,254 @@
+/-
+Copyright (c) 2021 Julian Kuelshammer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Kuelshammer
+-/
+import data.zmod.basic
+import group_theory.order_of_element
+import data.nat.basic
+import tactic.interval_cases
+import group_theory.dihedral
+
+/-!
+# Quaternion Groups
+
+We define the (generalised) quaternion groups `quaternion_group n` of order `4n`, also known as
+dicyclic groups, with elements `a i` and `xa i` for `i : zmod n`.
```

Comment:
The module docstring should not assume that readers know the definition.

Reply:
Thanks. Fixed.